tests/test_ecdsa.py

```diff
@@ -0,0 +1,82 @@
+V0_SIGNATURE_WITHOUT_VERSION = ('0x5d99b6f7f6d1f73d1a26497f2b1c89b24c0993913f86e9a2d02cd69887d9c94f3c880358579d811b'
+                                '21dd1b7fd9bb01c1d81d10e69f0384e675c32b39643be892')
+V1_SIGNER = '0x1E318623aB09Fe6de3C9b8672098464Aeda9100E'
+V1_SIGNATURE_WITHOUT_VERSION = ('0x331fe75a821c982f9127538858900d87d3ec1f9f737338ad67cad133fa48feff48e6fa0c18abc62'
+                                'e42820f05943e47af3e9fbe306ce74d64094bdf1691ee53e0')
+
+
+def test_recover_v0_sig_with_00_version(test_ecdsa_contract):
+    """test recover of v0 signature with 00 as version value"""
+    contract = test_ecdsa_contract
+    version = '00'
+    signature_with_version = V0_SIGNATURE_WITHOUT_VERSION + version
+    recovered_address = contract.functions.testRecover(TEST_MESSAGE,
+                                                       Web3.toBytes(hexstr=signature_with_version)).call()
+
+    assert recovered_address == V0_SIGNER
+
+
+def test_recover_v0_sig_with_27_version(test_ecdsa_contract):
+    """test recover of v0 signature with 27 as version value"""
+    contract = test_ecdsa_contract
+    version = '1b'  # 27 = 1b
+    signature_with_version = V0_SIGNATURE_WITHOUT_VERSION + version
+    recovered_address = contract.functions.testRecover(TEST_MESSAGE,
+                                                       Web3.toBytes(hexstr=signature_with_version)).call()
+
+    assert recovered_address == V0_SIGNER
+
+
+def test_recover_v0_sig_with_wrong_version(test_ecdsa_contract):
+    """test recover of v0 signature with wrong version value only valid version
+    values are 0, 1, 27, 28"""
+    contract = test_ecdsa_contract
+    version = '02'
+    signature_with_version = V0_SIGNATURE_WITHOUT_VERSION + version
+    recovered_address = contract.functions.testRecover(TEST_MESSAGE,
+                                                       Web3.toBytes(hexstr=signature_with_version)).call()
+
+    assert recovered_address == ZERO_ADDRESS
+
+
+def test_recover_v1_signature_01_version(test_ecdsa_contract):
+    """test recover of v0 signature with 01 as version value"""
+    contract = test_ecdsa_contract
+    version = '01'
+    signature_with_version = V1_SIGNATURE_WITHOUT_VERSION + version
+    recovered_address = contract.functions.testRecover(TEST_MESSAGE,
+                                                       Web3.toBytes(hexstr=signature_with_version)).call()
+
+    assert recovered_address == V1_SIGNER
+
+
+def test_recover_v1_signature_28_version(test_ecdsa_contract):
+    """test recover of v0 signature with 28 as version value"""
+    contract = test_ecdsa_contract
+    version = '1c'  # 28 = 1c
+    signature_with_version = V1_SIGNATURE_WITHOUT_VERSION + version
+    recovered_address = contract.functions.testRecover(TEST_MESSAGE,
+                                                       Web3.toBytes(hexstr=signature_with_version)).call()
+
+    assert recovered_address == V1_SIGNER
+
```

Comment:
We could probably add tests for signature with v1 and version 27, or signature with v0 and version 28 (both recoveries should return `ZERO_ADDRESS`). I do not know why OppenZeppelin did not do it.

Reply:
Hmm so just tried it out locally and the combinations of `V1_SIGNATURE` + `27` doesn't return `ZERO_ADDRESS`... Neither does `V0_SIGNATURE` + `28`. I think the contract only checks if the version value is either 0, 1, 27 or 28 but does not take the wrong combinations into account.

https://github.com/trustlines-network/contract-library/blob/162c0799e1f56757f6fae765926d1db18053bfe6/contracts/ECDSA.sol#L66-L72